### PR TITLE
handle non standard types in jdbc sources

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Some datatypes used by our JDBC sources are not "standard" SQL types and do not have type codes recognizable by `java.sql.JDBCTypes`. This causes an exception to be thrown when running `JDBCType.valueOf`. This PR catches this case and converts the type to string.

### CODE REVIEW

File: airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/MssqlSourceTest.java [hunk 1]
```diff
@@ -34,11 +34,15 @@
...
 import io.airbyte.protocol.models.AirbyteCatalog;
+import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.CatalogHelpers;
 import io.airbyte.protocol.models.Field;
 import io.airbyte.protocol.models.Field.JsonSchemaPrimitive;
 import io.airbyte.protocol.models.SyncMode;
 import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.swing.*;
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/MssqlSourceTest.java:45]: bad import

**Review by michel-tricot — APPROVED** [review]

File: airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/MssqlSourceTest.java [hunk 2]
```diff
@@ -108,6 +114,9 @@ void testDiscoverWithPk() throws Exception {
 
     final AirbyteCatalog actual = new MssqlSource().discover(config);
     assertEquals(CATALOG, actual);
+
+    List<AirbyteMessage> read = new MssqlSource().read(config, CatalogHelpers.toDefaultConfiguredCatalog(actual), null).collect(Collectors.toList());
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/MssqlSourceTest.java:118]: what is this doing / testing?

Comment (sherifnada) [reply airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/MssqlSourceTest.java:118]: oh crap. I accidentally added this to the diff.

File: airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java [hunk 3]
```diff
@@ -110,10 +110,18 @@ public static JsonNode rowToJson(ResultSet r) throws SQLException {
     return jsonNode;
   }
 
+  private static JDBCType safeGetJdbcType(int columnTypeInt) {
```
Comment (cgardens) [inline airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:113]: what was the type that was failing?

Comment (sherifnada) [reply airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:113]: `datetimeoffset` in mssql

**Pull request merged**